Empire/Empire.cpp

```diff
@@ -1357,6 +1357,10 @@ void Empire::AddNewlyResearchedTechToGrantAtStartOfNextTurn(const std::string& n
     auto result = m_newly_researched_techs.insert(name);
     if (result.second)
         AddSitRepEntry(CreateTechResearchedSitRep(name));
+
+    // Unlock items now as the SitReps get generated already for CurrentTurn()+1
+    for (const ItemSpec& item : tech->UnlockedItems())
```

Comment:
This partly undoes the change of moving tech adding to empires to after the turn number increment and effects, since those effects can depend on what stuff is unlocked. It is also strange to have separate paths for how to set when techs vs other stuff are unlocked.

Reply:
Well the root question is when do we want to have the unlocks and the effects available.

I think it is quite fine to have the effects start doing their effects in the turn after the "researched tech" message but the unlocks available in the turn of the  "researched tech" message.

The SitReps should align with that - so unlocks in Turn X and effects' SitReps (e.g. enemy detection increase) in Turn X+1.